Sentinel1 grd to rtc

MIRSL opened the pull request:
These updates mainly concern the rtcApp.py workflow for Sentinel 1 GRD data. It wouldn't run on my system originally, so I made some changes. Besides that, the main purpose was to add radiometric terrain correction (gamma 0) as an output. I think it works pretty well now, but please let me know any issues with these changes.

MIRSL committed:
Add files via upload

updated mainly to read the zipped datasets, remove orbit extender, try except for precise/coarse orbit

MIRSL committed:
Add files via upload

updated posting

MIRSL committed:
Add files via upload

MIRSL committed:
Add files via upload

updated for radiometric terrain correction
control looks by providing 'posting' in xml

MIRSL committed:
Update README.md

added example for running rtcApp

MIRSL committed:
Update README.md

MIRSL committed:
Update README.md

MIRSL committed:
Update README.md

MIRSL committed:
Update runNormalize.py

MIRSL committed:
Add files via upload

fixed issues relating to finding and selecting between orbits

MIRSL committed:
Update README.md

MIRSL committed:
Update README.md

piyushrpt commented on components/isceobj/RtcProc/runNormalize.py:
```diff
@@ -18,25 +20,29 @@ def runNormalize(self):
+                            beta = inname,
+                            out = outname,
+                            mask = maskname)
+        status = os.system(cmdrun)
```
Would it be possible to replace the os.system() command with direct calls to imageMath.main().
We are trying to not use system calls within the run methods. If not, we can add that as a separate PR. 

For separating the args and files:

```
class Dummy:
    pass

args = Dummy()
args.equation = 'a*cos(b_0*PI/180.)/cos(b_1*PI/180.) * (c==0)'
args.dtype = np.float32
args.scheme = 'BIL'
args.out = outname


files  = Dummy()
files.a = inname
files.b = incname
files.c =  maskname

imageMath.main(files, args)
```

MIRSL replied:
Thanks Piyush! That's no problem, I will look into that soon ... thanks for letting me know.

MIRSL replied:
Hi Piyush, I just updated the codes to avoid the system call as you suggested. I also slightly modified imageMath.py because I encountered some issues otherwise. Thank you!

MIRSL committed:
Add files via upload

MIRSL committed:
Add files via upload

changed to use imagemath.main() instead of command line

MIRSL committed:
Add files via upload

couple edits to deal with receiving data as list (the command line/system call approach) versus receiving data passed as object

piyushrpt commented:
Looks good to me. Let me know if this works and that you have tested it. We can merge.

MIRSL commented:
Hi Piyush, I've tested this workflow (rtcApp.py) and it works fine! Thank you!

piyushrpt commented:
Oh ... can you add a simple test to .circleci/config.yml  just like 

```
rtcApp.py --help --steps
```

https://github.com/isce-framework/isce2/blob/master/.circleci/config.yml#L79-L80
